### PR TITLE
Content changes for opposite and same sex outcomes where user lives in UK or SA

### PR DESCRIPTION
**Pull request by ikennaokpala**
Trello Story: https://trello.com/c/SSdBWMvO/107-marriage-abroad-south-africa

## Factcheck
* Opposite sex outcomes where user is planning on getting married in South Africa and is resident in the UK:
   * [Preview Link](https://smart-answers-pr-2467.herokuapp.com/marriage-abroad/y/south-africa/uk/partner_british/opposite_sex)
   * [Link on GOV.UK](https://gov.uk/marriage-abroad/y/south-africa/uk/partner_british/opposite_sex)

## Expected changes
* Content changes for opposite sex outcomes where user is resident in the UK
  * Changes to Legalisation and translation

### Before
screen shot 2016-04-18 at 11 53 59

### After
screen shot 2016-04-18 at 11 53 51

## Factcheck
* Same sex outcomes where user is planning on getting married in South Africa and is resident in the UK:
   * [Preview Link](https://smart-answers-pr-2467.herokuapp.com/marriage-abroad/y/south-africa/uk/partner_british/same_sex)
   * [Link on GOV.UK](https://gov.uk/marriage-abroad/y/south-africa/uk/partner_british/same_sex)

## Expected changes
* Content changes for opposite sex outcomes where user is resident in the UK
  * Changes to Legalisation and translation

### Before
screen shot 2016-04-18 at 11 57 13

### After
screen shot 2016-04-18 at 11 57 40

## Factcheck
* Opposite sex outcomes where user is planning on getting married in South Africa and is resident in the South Africa:
   * [Preview Link](https://smart-answers-pr-2467.herokuapp.com/marriage-abroad/y/south-africa/ceremony_country/partner_local/opposite_sex)
   * [Link on GOV.UK](https://gov.uk/marriage-abroad/y/south-africa/ceremony_country/partner_local/opposite_sex)

## Expected changes
* Content changes for opposite sex outcomes where user is resident in the South Africa
  * Changes to Legalisation and translation

### Before
screen shot 2016-04-18 at 12 00 33

### After
screen shot 2016-04-18 at 12 00 20

## Factcheck
* Same sex outcomes where user is planning on getting married in South Africa and is resident in the South Africa:
   * [Preview Link](https://smart-answers-pr-2467.herokuapp.com/marriage-abroad/y/south-africa/ceremony_country/partner_local/same_sex)
   * [Link on GOV.UK](https://gov.uk/marriage-abroad/y/south-africa/ceremony_country/partner_local/same_sex)

## Expected changes
* Content changes for opposite sex outcomes where user is resident in the South Africa
  * Changes to Legalisation and translation

### Before

screen shot 2016-04-18 at 12 02 29

### After

screen shot 2016-04-18 at 12 02 52

### CODE REVIEW

File: lib/smart_answer_flows/marriage-abroad.rb [hunk 1]
```diff
@@ -126,6 +126,13 @@ def define
           elsif calculator.partner_is_opposite_sex?
             if calculator.ceremony_country == 'hong-kong'
               outcome :outcome_opposite_sex_marriage_in_hong_kong
+            elsif calculator.ceremony_country == 'south-africa' &&
+                (
+                  calculator.resident_of_uk? ||
+                  calculator.resident_of_ceremony_country? ||
+                  calculator.resident_of_third_country?
+                )
```
Comment (leenagupte) [inline lib/smart_answer_flows/marriage-abroad.rb]: Is this check needed? A user must always be either a resident of the UK, a resident of South Africa or a resident of another country, so this check will always be truthy.

File: lib/smart_answer_flows/marriage-abroad.rb [hunk 2]
```diff
@@ -197,6 +204,13 @@ def define
           elsif calculator.partner_is_same_sex?
             if %w(belgium norway).include?(calculator.ceremony_country)
               outcome :outcome_same_sex_civil_partnership_in_affirmation_countries
+            elsif calculator.ceremony_country == 'south-africa' &&
+                (
+                  calculator.resident_of_uk? ||
+                  calculator.resident_of_ceremony_country? ||
+                  calculator.resident_of_third_country?
+                )
+              outcome :outcome_opposite_same_sex_marriage_residing_in_uk_or_south_africa
```
Comment (leenagupte) [inline lib/smart_answer_flows/marriage-abroad.rb]: The checks for a partner of the opposite sex or same sex send you to the same outcome. I think you should consolidate these into one statement that just checks the value of the ceremony_country is 'south-africa'

File: lib/smart_answer_flows/marriage-abroad.rb (hunk 1 above)
Comment (ikennaokpala) [inline lib/smart_answer_flows/marriage-abroad.rb]: @leenagupte this check initially started with like with the require for user either a uk resident or resident of ceremony_country

```
elsif calculator.ceremony_country == 'south-africa' &&
      (
         calculator.resident_of_uk? ||
         calculator.resident_of_ceremony_country? 
      )
```

After factcheck FCO made another requirement for third_country

I have decided to keep it this way for explicit reasons, make it a bit more obvious the 3 potential areas affected by this node.

File: lib/smart_answer_flows/marriage-abroad.rb (hunk 2 above)
Comment (ikennaokpala) [inline lib/smart_answer_flows/marriage-abroad.rb]: SEE [here](https://github.com/alphagov/smart-answers/pull/2467#discussion_r61056915)

**Comment by leenagupte** [comment]
Apart from my comments, LGTM.

File: lib/smart_answer_flows/marriage-abroad.rb (hunk 1 above)
Comment (floehopper) [inline lib/smart_answer_flows/marriage-abroad.rb]: I agree with @leenagupte about removing this check. I think it makes the code more complex for no real benefit.

**Pull request merged**